da/celestia/celestia.go

```diff
@@ -202,24 +202,16 @@ func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 				continue
 			}
 
+			//double check txResponse is not nil - not supposed to happen
 			if txResponse == nil {
-				c.logger.Error("Failed to submit DA batch. Emitting health event and trying again", "error", "txResponse is nil")
-				res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, false, errors.New("txResponse is nil"))
-				if err != nil {
-					return res
+				err := errors.New("txResponse is nil")
+				c.logger.Error("Failed to submit DA batch", "error", err)
+				return da.ResultSubmitBatch{
```

Comment:
why are we returning here and not retrying and submitting an health event? the node will simply panic.

Reply:
well it's not supposed to happen, so IMO panic is the right choice (similar to invariant break)
WDYT

Reply:
I don't see it as an invariant broken or a bug so I don't think we should panic, especially if retry could help (which is why we added it in the previous version as it was the celestia's network instability/response).

Also if we panic and restart don't see why would it change anything if it depends on the celestia network.

Reply:
it's not depend on the network
it's part of the API contract. u don't suppose to get nil resp and nil err

but ok I'll change it